File: face_detection/ZeticMLangeFaceDetection-Android/app/src/main/java/com/zeticai/facedetection_android/feature/FaceDetection.kt
package com.zeticai.facedetection_android.feature

import android.content.Context
import com.zetic.ZeticMLange.ZeticMLangeModel
import com.zetic.ZeticMLangeFeature.ZeticMLangeFeatureFaceDetection
import com.zetic.ZeticMLangeFeature.type.FaceDetectionResult

class FaceDetection @JvmOverloads constructor(
    context: Context,
    modelKey: String,
    private val model: ZeticMLangeModel = ZeticMLangeModel(context, modelKey),
    private val featureModel: ZeticMLangeFeatureFaceDetection = ZeticMLangeFeatureFaceDetection()
) {
    fun run(imagePtr: Long): FaceDetectionResult {
        val preprocess = featureModel.preprocess(imagePtr)
        model.run(arrayOf(preprocess))
        val modelOutput = model.outputBuffers
        return featureModel.postprocess(modelOutput.map {
            val byteArray = ByteArray(it.remaining())
            it.get(byteArray)
            return@map byteArray
        }.toTypedArray())
    }

    fun close() {
        model.deinit()
        featureModel.deinit()
    }
}

File: face_landmark/ZeticMLangeFaceLandmark-iOS/ZeticMLangeFaceLandmark-iOS/FaceDetection.swift
import Foundation
import UIKit
import ZeticMLange

class FaceDetection {
    private let model: ZeticMLangeModel
    private let featureModel: ZeticMLangeFeatureFaceDetection
    
    init (_ modelKey: String) {
        model = ZeticMLangeModel(modelKey)!
        featureModel = ZeticMLangeFeatureFaceDetection()
    }
    
    func run(_ image: UIImage) -> Array<FaceDetectionResult> {
        do {
            let preprocess = featureModel.preprocess(image)
            try model.run([preprocess])
            var modelOutput = model.getOutputDataArray()
            return featureModel.postprocess(&modelOutput)
        } catch {
            print(error)
            return []
        }
    }
}
